File: rust/src/meetup_version.rs
// Disclaimer: Horrible code produced under time pressure...

use std::env;
use std::fs::File;
use std::io::{BufReader, BufRead};
use std::str::FromStr;

#[derive(Debug)]
struct Command {
    column: usize,
    op: Op,
}

#[derive(Debug)]
enum Op {
    Sum,
    Average,
}

impl FromStr for Command {
    type Err = ();
    fn from_str(input: &str) -> Result<Command, ()> {
        let mut parts = input.split('=');
        let col = parts.next()
            .unwrap()
            .parse()
            .unwrap();
        let op = match parts.next().unwrap() {
            "SUM" => Op::Sum,
            "MEAN" => Op::Average,
            _ => panic!("Op parse error!"),
        };
        Ok(Command {
               column: col,
               op: op,
           })

    }
}

fn main() {
    let mut args = env::args().skip(1);
    let file_path = args.next().expect("Missing file!");
    let mut commands: Vec<Command> = Vec::new();
    for argument in args {
        commands.push(argument.parse().unwrap());
    }

    let file = File::open(file_path).unwrap();
    let mut lines = BufReader::new(file).lines();

    let first_line = lines.next().unwrap().unwrap();
    let columns = first_line.split(',').count();
    let mut column_sums: Vec<u32> = Vec::with_capacity(columns);
    column_sums.resize(columns, 0);

    let mut rows = 0;
    for line in lines {
        rows += 1;
        let line = line.unwrap();
        let fields = line.split(',');
        for (col, field) in fields.enumerate() {
            column_sums[col] += field.parse().expect("ParseError");
        }
    }

    for command in commands {
        let col = command.column;
        match command.op {
            Op::Sum => {
                println!("{}[SUM]: {}", col - 1, column_sums[col - 1]);
            }
            Op::Average => {
                println!("{}[MEAN]: {}",
                         col - 1,
                         column_sums[col - 1] as f64 / rows as f64);
            }
        }
    }
}

File: py/meetup_version.py
#!/usr/bin/env python

from sys import argv

mode = {'MEAN': 1, 'SUM': 2}

def parse(arg, col, mode):
    if mode not in [1, 2]:
        return
    with open(arg, 'r') as csv_input:
        sum = 0
        first_line = csv_input.readline()
        if col > len(first_line.split(',')) - 1:
            return
        for idx, line in enumerate(csv_input.readlines()):
            if idx == 0:
                continue
            values = line.split(",")
            sum += int(values[col])
        if mode == 1:
            print(col, "[MEAN]:", sum/idx)
        elif mode == 2:
            print(col, "[SUM]:", sum)

if __name__ == "__main__":
    if len(argv) < 2:
        print("Argument error.")
        exit(-1)
    for arg in argv[2:]:
        param = arg.split('=')
        col = int(param[0])
        m = mode[param[1]]
        parse(argv[1], col, m)
